src/polymer/polymer-element.ts

```diff
@@ -262,9 +262,9 @@ function resolveElement(
     scannedElement: ScannedPolymerElement, document: Document): PolymerElement {
   const element = new PolymerElement();
   element.privacy = scannedElement.privacy;
-  applySuperClass(element, scannedElement, document);
-  applyMixins(element, scannedElement, document);
   applySelf(element, scannedElement, document);
+  applyMixins(element, scannedElement, document);
+  applySuperClass(element, scannedElement, document);
```

Comment:
Basically the naive method would be to walk the array for each inherited property to see if it should overwrite any entries in the array. To do O(N) we'd need to maintain an additional side-datastructure for doing this quickly. If we only append and overwrite to the array this could just be a Map<string, number>, so maybe this won't be so bad.